### ISSUE TITLE
Support for early stopping in ChainerMN

### ISSUE THREAD

**Issue by shu65** [chainer/chainer#6616]
Chainer has 'EarlyStoppingTrigger' to support early stopping.
It is not able to be used when ChainerMN is used.

But, the support for it in ChainerMN may be difficult, because the additional communications are required for it.  
Does anyone have any idea to solve this problem?

**Comment by shu65** [comment]
@kuenishi 's idea:
Instead of communicating every iteration, mpi workers communicate every 10 iterations and every 100 iterations to judge

**Comment by shmurai** [comment]
With #7302, I could realize early stopping in ChainerMN using the code like:

```
trigger = training.triggers.EarlyStoppingTrigger(monitor='main/loss-ave', max_trigger=(args.epoch, 'epoch'))
trainer = training.Trainer(updater, trigger, out=args.out)
trainer.extend(chainermn.extensions.observation_aggregator(comm, 'main/loss', 'main/loss-ave'))
```

Although I think this is simple enough, it may be better to add a function to do this all at once.

**Comment by shmurai** [comment]
MultiNodeEarlyStoppingTrigger was implemented in #7302 along with ObservationAggregator.
